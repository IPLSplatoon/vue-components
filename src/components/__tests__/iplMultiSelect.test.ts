import IplMultiSelect from '../iplMultiSelect.vue';
import { mount } from '@vue/test-utils';

describe('IplMultiSelect', () => {
    it('applies given label to element', () => {
        const wrapper = mount(IplMultiSelect, { props: { label: 'Label!', modelValue: [], options: []} });

        expect(wrapper.get('label').text()).toEqual('Label!');
    });

    it('applies given options to select element', () => {
        const wrapper = mount(IplMultiSelect, {
            props: {
                label: '',
                modelValue: [],
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        expect(wrapper.get('select').html()).toMatchSnapshot();
    });

    it('creates element for each selected option', () => {
        const wrapper = mount(IplMultiSelect, {
            props: {
                label: '',
                modelValue: [{ name: 'Option', value: 'opt' }, { name: 'Opt 3', value: 'optthree' }],
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        expect(wrapper.get('.elem-display').html()).toMatchSnapshot();
    });

    it('emits update message on change', async () => {
        const wrapper = mount(IplMultiSelect, {
            props: {
                label: '',
                modelValue: [],
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        await wrapper.get('select').setValue('opt2');

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted?.length).toEqual(1);
        expect(emitted?.[0]).toEqual([[{ name: 'Option the Second', value: 'opt2' }]]);
    });

    it('handles selected option being removed', async () => {
        const wrapper = mount(IplMultiSelect, {
            props: {
                label: '',
                modelValue: [{ name: 'Option', value: 'opt' }, { name: 'Option the Second', value: 'opt2' }],
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        await wrapper.get('.option').trigger('click');

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted?.length).toEqual(1);
        expect(emitted?.[0]).toEqual([[{ name: 'Option the Second', value: 'opt2' }]]);
    });

    it('removes selected option when it is removed from options list', async () => {
        const wrapper = mount(IplMultiSelect, {
            props: {
                label: '',
                modelValue: [{ name: 'Option', value: 'opt' }, { name: 'Option the Second', value: 'opt2' }],
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        await wrapper.setProps({
            options: [
                { name: 'Option', value: 'opt' },
                { name: 'Opt 3', value: 'optthree' }
            ]
        });

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted?.length).toEqual(1);
        expect(emitted?.[0]).toEqual([[{ name: 'Option', value: 'opt' }]]);
    });
});
