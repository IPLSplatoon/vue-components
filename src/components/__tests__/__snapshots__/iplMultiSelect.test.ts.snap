// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplMultiSelect applies given options to select element 1`] = `
"<select>
  <option value=\\"opt\\">Option</option>
  <option value=\\"opt2\\">Option the Second</option>
  <option value=\\"optthree\\">Opt 3</option>
</select>"
`;

exports[`IplMultiSelect creates element for each selected option 1`] = `
"<div class=\\"elem-display\\">
  <div class=\\"option\\">Option</div>
  <div class=\\"option\\">Opt 3</div>
</div>"
`;
